metaflow/datastore/task_datastore.py

```diff
@@ -435,6 +435,37 @@ def save_metadata(self, contents, allow_overwrite=True, add_attempt=True):
             add_attempt,
         )
 
+    @require_mode("w")
+    def _dangerous_save_metadata_post_done(
+        self, contents, allow_overwrite=True, add_attempt=True
+    ):
+        """
+        Method identical to save_metadata BUT BYPASSES THE CHECK ON DONE
```

Comment:
Should .save_metadata() call this method then? Or maybe just remove `@only_if_done` decorator? I'm not 100% sure what scenario is supposed to prevent though, or is it just a sanity check?

Also consider adding a comment explaining what exactly is the "danger" here, I understand that it bypasses that check but what's the actual consequence of this?

Reply:
I put some additional comment. Given the length of the method, I rather keep them separate. It is also a little nicer not to call a _dangerous method inside a non dangerous one :).